src/app/pages/ArticlePage/ContinueReadingButton/index.tsx

```diff
@@ -37,51 +37,45 @@ const ContinueReadingButton = ({
   const { onClick: clickTrackerHandler } =
     useClickTrackerHandler(eventTrackingData);
 
+  useEffect(() => {
+    if (showAllContent && firstHiddenElement) {
+      // Apply the custom focus style dynamically
+      firstHiddenElement.tabIndex = 0;
+      firstHiddenElement.focus();
+    }
+  }, [firstHiddenElement, showAllContent]);
+
   const handleEvent = (event: MouseEvent<HTMLButtonElement>) => {
     clickTrackerHandler?.(event);
-    setShowAllContent();
-  };
 
-  useEffect(() => {
-    if (showAllContent) {
+    const maybeKeyboardEvent = event.detail === 0;
```

Comment:
React reports keyboard/mouse/tap as `click` in `event.type`, so this is a bit of a hacky, but fairly reliable way of determining if the event is a keyboard event, without needing separate `onKeyPress` style handlers.